### PR TITLE
lcarr epl sandbox proj

### PR DESCRIPTION
**Pull request by lindsayplatt**

### CODE REVIEW

File: epl_trends/process_json_data.js [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+	// setup line function
+	var buildLine = d3.line().curve(d3.curveCardinal)
+		.x(function(d) { return xScale(d.startyear); })
+		.y(function(d) { return yScale(d[yvar]); });
+
+	// organize data by team
+	var databyteam = d3.nest()
+    	.key(function(d) { return d.team; })
+    	.entries(data);
```
Comment (aappling-usgs) [inline epl_trends/process_json_data.js:107]: wowee! impressive!

File: epl_trends/process_json_data.js [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+	var lines_g = plot.append("g").attr("class", "teamline")	
+	databyteam.forEach(function(d, i) {
+		lines_g.append("path")
+		.attr("class", "teamline")
+		.attr("d", buildLine(d.values))
+		.style("stroke", colorScale(d.key))
+		.style("fill", "none");
+	})
+
+	// update phase????
```
Comment (aappling-usgs) [inline epl_trends/process_json_data.js]: yes, i think so. was this part working?

**Comment by lindsayplatt** [comment]
exit phase still not working

File: epl_trends/process_json_data.js [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+		
+	// add lines
+	var lines_g = plot.append("g").attr("class", "teamline")	
+	databyteam.forEach(function(d, i) {
+		lines_g.append("path")
+		.attr("class", "teamline")
+		.attr("d", buildLine(d.values))
+		.style("stroke", colorScale(d.key))
+		.style("fill", "none");
+	})
```
Comment (aappling-usgs) [inline epl_trends/process_json_data.js]: i'm probably missing something important, and am not testing this, but maybe something like:
```
var lines = lines_g.
  .data(databyteam)
  .append("path")
  .attr("class", "teamline")
  .attr("d", function(d) { return buildLine(d.values) })
  .etc.
```
does that work? seems like `.data()` and selections ought to be useful for iterating over elements of databyteam

File: epl_trends/process_json_data.js [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+			   "04-05","05-06","06-07","07-08","08-09","09-10",
+			   "10-11","11-12","12-13","13-14","14-15"];
+	//"https://raw.githubusercontent.com/jokecamp/FootballData/master/EPL%201992%20-%202015/tables/epl-92-93.json"
+	var baseurl = "https://raw.githubusercontent.com/jokecamp/FootballData/master/EPL%201992%20-%202015/tables/epl";
+	//console.log(seasons);
+	var q = d3.queue();
+	seasons.forEach(function(s) {
+		url = baseurl + "-" + s + ".json";
+		q.defer(parseEPLJson, url, s);
+	});
```
Comment (aappling-usgs) [inline epl_trends/process_json_data.js:18]: awesome

File: epl_trends/process_json_data.js [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+
+// custom function to read json files and add the seasons to the json data
+function parseEPLJson(url, season, callback) {
+	d3.json(url, function(data) {
+		data.forEach(function(d) {
+			d.season = season;
+			d["goals-dff"] = parseFloat(d["goals-dff"]); //parseFloat for handling negative values
+			// keep only the first season and convert to year
+				var yr1 = season.match(/[0-9]{2}-*/)[0];
+				d.startyear = d3.timeParse("%y")(yr1.replace("-","")).getFullYear();
```
Comment (aappling-usgs) [inline epl_trends/process_json_data.js]: so many gems in here! thanks for sharing!

File: epl_trends/process_json_data.js [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+        }
+
+        plotarea = addDataToPlot(plotarea, merged_data, new_yvar);
+        
+        //d3.selectAll(".datapt").exit().remove();
+		//d3.selectAll(".teamline").exit().remove();
+		console.log(d3.selectAll(".datapt"));
+
+		//by default html form submissions does a full page refresh, this turns that off
+        d3.event.preventDefault();
```
Comment (aappling-usgs) [inline epl_trends/process_json_data.js]: :+1:

**Review by aappling-usgs — APPROVED** [review]
Rock on, lindsay!

**Comment by aappling-usgs** [comment]
I think you need to add an update phase that re-sets the `d` attribute - not many of your lines are actually exiting in D3's eyes, most are updating.

**Comment by lindsayplatt** [comment]
OK, I think these most recent changes do what you are saying @aappling-usgs. Having an issue with everything resetting to the original plot every time now....

**Comment by lindsayplatt** [comment]
Alright - things are working!!!! Much much nicer than before. Definitely still has room for improvements, but I'm satisfied with merging this :)

**Pull request merged**